server/channels/app/bot.go

```diff
@@ -124,7 +124,9 @@ func (a *App) CreateBot(rctx request.CTX, bot *model.Bot) (*model.Bot, *model.Ap
 
 	savedBot, nErr := a.Srv().Store().Bot().Save(bot)
 	if nErr != nil {
-		a.Srv().Store().User().PermanentDelete(rctx, bot.UserId)
+		if err := a.Srv().Store().User().PermanentDelete(rctx, bot.UserId); err != nil {
+			a.Log().Error("Failed to permanently delete user", mlog.Err(err))
```

Comment:
Let's use the request logger
```suggestion
			rctx.Logger().Error("Failed to permanently delete user", mlog.Err(err))
```